core/src/main/java/com/sequenceiq/cloudbreak/service/AbstractWorkspaceAwareResourceService.java

```diff
@@ -76,6 +78,21 @@ public T getByNameForWorkspace(String name, Workspace workspace) {
         return object;
     }
 
+    @Override
+    public Set<T> getByNamesForWorkspaceId(Set<String> names, Long workspaceId) {
+        Set<T> results = repository().findByNameInAndWorkspaceId(names, workspaceId);
+        Set<String> notFound = Sets.difference(names,
+                results.stream().map(WorkspaceAwareResource::getName).collect(Collectors.toSet()));
+
+        if (!notFound.isEmpty()) {
+            throw new NotFoundException(String.format("No %s(s) found with name(s) '%s'", resource().getShortName(),
+                    String.join(", ",
+                            notFound.stream().map(name -> "'" + name + "'").collect(Collectors.toSet()))));
```

Comment:
I suggest to use `throw new NotFoundException(String.format("No %s(s) found with name(s) '%s'", resource().getShortName(),
                    notFound.stream().map(name -> '\'' + name + '\'').collect(Collectors.joining(", "))));` in this case, because not need to collext elements into a `Set`